Source/Core/DolphinQt2/NetPlay/PadMappingDialog.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/NetPlay/PadMappingDialog.h"
+
+#include "Core/NetPlayClient.h"
+#include "DolphinQt2/NetPlay/NetPlay.h"
+
+#include <QComboBox>
+#include <QDialogButtonBox>
+#include <QGridLayout>
+#include <QLabel>
+
+PadMappingDialog::PadMappingDialog(QWidget* parent) : QDialog(parent)
+{
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+  CreateWidgets();
+  ConnectWidgets();
+}
+
+void PadMappingDialog::CreateWidgets()
+{
+  m_main_layout = new QGridLayout;
+  m_button_box = new QDialogButtonBox(QDialogButtonBox::Ok);
+
+  for (unsigned int i = 0; i < m_wii_boxes.size(); i++)
+  {
+    m_gc_boxes[i] = new QComboBox;
+    m_wii_boxes[i] = new QComboBox;
+
+    m_main_layout->addWidget(new QLabel(tr("GC Pad %1").arg(i + 1)), 0, i);
+    m_main_layout->addWidget(new QLabel(tr("Wiimote %1").arg(i + 1)), 0, 4 + i);
+    m_main_layout->addWidget(m_gc_boxes[i], 1, i);
+    m_main_layout->addWidget(m_wii_boxes[i], 1, 4 + i);
+  }
+
+  m_main_layout->addWidget(m_button_box, 2, 7);
+
+  setLayout(m_main_layout);
+}
+
+void PadMappingDialog::ConnectWidgets()
+{
+  connect(m_button_box, &QDialogButtonBox::accepted, this, &QDialog::accept);
+}
+
+int PadMappingDialog::exec()
+{
+  m_players = QtNetPlay::GetClient()->GetPlayers();
+
+  QStringList players;
+
+  players.append(tr("None"));
+
+  for (const auto& player : m_players)
+    players.append(QString::fromStdString(player->name));
+
+  for (unsigned int i = 0; i < m_wii_boxes.size(); i++)
```

Comment:
There's nothing in the code that guarantees m_wii_boxes and m_gc_boxes to be the same size. To be safe, and cause it's easy here, this is probably better:

```c++
for (auto& combo_group :{m_gc_boxes, m_wii_boxes})
{
  for (auto& combo : combo_group)
  {
    // ... 
  }
};
```